src/utils/request.js

```diff
@@ -20,35 +20,78 @@ const codeMessage = {
   504: 'The gateway timed out. ',
 };
 
+const customMessages = new Set([
+  'A user with that name already exists.',
+  'No such user, please register first',
+  'Bad login',
```

Comment:
It bothers me to have the dashboard keying on exact message text; it'll break if we change wording for any reason. (And I presume that at some point we'll be adding message localization.)

A lot of error message responses have an "ID" string that can be more stable... we ought to give that some thought. E.g., the response might be something like this

```
{
    "id": "DUPLICATE_USER",
    "message": "A user with that name already exists."
}
```

and the dashboard could recognize the "known" errors by id and use whatever message text came with it.

Or if you think there's some other server change that would make this easier for you, let's talk about it! Don't treat the server as an immutable external entity; I really want our API to be generally usable for scripting (or even alternate UI integration), but this sort of thing is relevant for any API client anyway.

Reply:
As it is with `umi-request` we need to conditionally compare against a known set of messages for user authentication to handle errors in some other way. So, when we get a response from the server, we need to compare that with some known set of values to render something different that default (throwing exception).

Yes, using an ID with the response can help immensely in this case.